### PR TITLE
chore: update CHANGELOG.md for v0.31.0

### PR DESCRIPTION
**Pull request by Ivansete-status**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,55 @@
...
+- update content-topic parsing for filter (#2835) ([733edae43](https://github.com/waku-org/nwaku/commit/733edae43))
+- better descriptive log (#2826) ([94947a850](https://github.com/waku-org/nwaku/commit/94947a850))
+- zerokit: bump submodule (#2830) ([c483acee3](https://github.com/waku-org/nwaku/commit/c483acee3))
+
+This release supports the following [libp2p protocols](https://docs.libp2p.io/concepts/protocols/):
+| Protocol | Spec status | Protocol id |
+| ---: | :---: | :--- |
+| [`11/WAKU2-RELAY`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/11/relay.md) | `stable` | `/vac/waku/relay/2.0.0` |
+| [`12/WAKU2-FILTER`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/12/filter.md) | `draft` | `/vac/waku/filter/2.0.0-beta1` <br />`/vac/waku/filter-subscribe/2.0.0-beta1` <br />`/vac/waku/filter-push/2.0.0-beta1` |
+| [`13/WAKU2-STORE`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/13/store.md) | `draft` | `/vac/waku/store/2.0.0-beta4` |
```
Comment (SionoiS) [inline CHANGELOG.md:48]: do we change this to `/vac/waku/store/3.0.0` or not yet?

**Review by SionoiS — APPROVED** [review]

**Review by NagyZoltanPeter — APPROVED** [review]
Looks just good from my side!

File: CHANGELOG.md (hunk 1 above)
Comment (Ivansete-status) [reply CHANGELOG.md:48]: > do we change this to `/vac/waku/store/3.0.0` or not yet?

ah yes, good point! In that case, it should be `/vac/waku/store-query/3.0.0` right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+- duplicate message forwarding in filter service (#2842) ([99149ea9d](https://github.com/waku-org/nwaku/commit/99149ea9d))
+- only set disconnect time on left event (#2831) ([01050138c](https://github.com/waku-org/nwaku/commit/01050138c))
+- adding peer exchange peers to the peerStore (#2824) ([325e13169](https://github.com/waku-org/nwaku/commit/325e13169))
+- ci use --tags to match non-annotated tags (#2814) ([317c83dc1](https://github.com/waku-org/nwaku/commit/317c83dc1))
+- update peers ENRs in peer store in case they are updated (#2818) ([cda18f96c](https://github.com/waku-org/nwaku/commit/cda18f96c))
+- mount metadata in wakucanary (#2793) ([3b27aee82](https://github.com/waku-org/nwaku/commit/3b27aee82))
+
+### Changes
+
+- enhance postgres and retention policy logs (#2884) ([71ee42de5](https://github.com/waku-org/nwaku/commit/71ee42de5))
```
Comment (gabrielmer) [inline CHANGELOG.md:23]: Adding https://github.com/waku-org/nwaku/pull/2914 which will enter the release :)


```suggestion

- setting filter handling logs to trace (#2914) ([5c539fe13](https://github.com/waku-org/nwaku/commit/5c539fe13))
- enhance postgres and retention policy logs (#2884) ([71ee42de5](https://github.com/waku-org/nwaku/commit/71ee42de5))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,56 @@
+## v0.31.0 (2024-07-16)
+
+### Features
```
Comment (gabrielmer) [inline CHANGELOG.md:3]: I think we should add a note that we officially deprecated named sharding

```suggestion
### Notes

* Named sharding has been deprecated in favor of static sharding. Topics in formats other than `/waku/2/rs/<cluster>/<shard>` are no longer supported

### Features
```

**Review by gabrielmer — APPROVED** [review]
Thank you! Proposed a couple additions :))

**Pull request merged**